processor_function/shared/cnab_builder.py

```diff
@@ -29,22 +29,41 @@ def __init__(self, resource_request_message: dict):
         self._container_group_name = ""
         self._location = ""
 
-    def _build_porter_cmd_line(self) -> str:
+    def _build_porter_cmd_line(self) -> List[str]:
         porter_parameters = ""
         for key in self._message['parameters']:
             porter_parameters += " --param " + key + "=" + self._message['parameters'][key]
 
-        start_command_line = "/bin/bash -c porter " + self._message['operation'] + " CNAB --tag " + self._message[
-            'bundle-name'] + porter_parameters + " -d azure && porter show CNAB"
-
-        logging.info("Creating a runner with:" + start_command_line)
+        for key, value in os.environ.items():
+            if key.startswith("param_"):
+                porter_parameters += " --param " + key.replace('param_', '') + "=" + value
+
+        installation_id = self._message['parameters']['tre_id'] + "-" + self._message['parameters']['workspace_id']
+        start_command_line = ["/bin/bash", "-c", "az login --identity "
+                              + "&& TOKEN=$(az acr login --name msfttreacr --expose-token --output tsv --query accessToken) "
+                              + "&& docker login msfttreacr.azurecr.io --username 00000000-0000-0000-0000-000000000000 --password $TOKEN "
+                              + "&& porter "
+                              + self._message['action'] + " "
+                              + installation_id
+                              + " --reference "
+                              + os.environ["REGISTRY_SERVER"]
+                              + os.environ["WORKSPACES_PATH"]
+                              + self._message['name'] + ":"
+                              + "v" + self._message['version'] + " "
+                              + porter_parameters
+                              + " --cred ./home/porter/azure.json"
+                              + " -d azure && porter show " + installation_id]
+
+        # start_command_line = ["/bin/bash", "-c", "sleep 600000000"]
         return start_command_line
 
     def _build_cnab_env_variables(self) -> List[str]:
```

Comment:
This whole block could be much simpler.
Extract creating of EnvironmentVariable into a function which can return secure or non secure
Then you can use liscomp instead of for loop. It makes the intention clear and makes it look more pythonic

Reply:
changed and test added